addOns/pscanrules/src/test/java/org/zaproxy/zap/extension/pscanrules/TestInfoSessionIdURLUnitTest.java

```diff
@@ -0,0 +1,427 @@
+import org.zaproxy.zap.utils.ZapXmlConfiguration;
+
+public class TestInfoSessionIdURLUnitTest extends PassiveScannerTest<TestInfoSessionIdURL> {
+
+    private HttpMessage msg;
+    private static final String BODY = "Some text in the response, doesn't matter.\nLine 2\n";
+
+    @Override
+    protected TestInfoSessionIdURL createScanner() {
+
+        TestInfoSessionIdURL scanner = new TestInfoSessionIdURL();
+        //        scanner.setConfig(new ZapXmlConfiguration());
+        setUpHttpSessionsParam();
+        return scanner;
+    }
+
+    protected HttpMessage createHttpMessageWithRespBody(String responseBody)
+            throws HttpMalformedHeaderException, URIException {
+
+        HttpRequestHeader requestHeader = new HttpRequestHeader();
+        requestHeader.setURI(new URI("http://example.com", false));
+
+        msg = new HttpMessage();
+        msg.setRequestHeader(requestHeader);
+        msg.setResponseBody(responseBody);
+        msg.setResponseHeader(
+                "HTTP/1.1 200 OK\r\n"
+                        + "Server: Apache-Coyote/1.1\r\n"
+                        + "Content-Type: text/plain\r\n"
+                        + "Content-Length: "
+                        + responseBody.length()
+                        + "\r\n");
+        return msg;
+    }
+
+    @Test
+    public void shouldHaveSessionIdsInConfig() throws Exception {
+
+        // Given
+        OptionsParam options = Model.getSingleton().getOptionsParam();
+        HttpSessionsParam sessionOptions = options.getParamSet(HttpSessionsParam.class);
+
+        // When
+        List<String> sessionIds = new ArrayList<String>();
+        if (sessionOptions != null) {
+            sessionIds = sessionOptions.getDefaultTokensEnabled();
+        }
+
+        // Then
+        assertThat(sessionIds, is(not(empty())));
+    }
+
+    @Test
+    public void containsJSESSIONIDAsUrlParameter()
+            throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String testURI = "http://example.com/foo?jsessionid=1A530637289A03B07199A44E8D531427";
+        HttpMessage msg = createHttpMessageWithRespBody(BODY);
+        msg.getRequestHeader().setURI(new URI(testURI, false));
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertEquals(1, alertsRaised.size());
+    }
+
+    @Test
+    public void containsJSESSIONIDAsUrlParameterInHTTPS()
+            throws HttpMalformedHeaderException, URIException {
+
+        // Given
+        String testURI = "https://example.com/foo?jsessionid=1A530637289A03B07199A44E8D531427";
+        HttpMessage msg = createHttpMessageWithRespBody(BODY);
+        msg.getRequestHeader().setURI(new URI(testURI, false));
+
+        // When
+        rule.scanHttpRequestSend(msg, -1);
+
+        // Then
+        assertEquals(1, alertsRaised.size());
+    }
+
+    @Test
+    public void noSessionIdInHTTPS() throws HttpMalformedHeaderException, URIException {
```

Comment:
The scheme should be removed or added a test for http, the scheme is not relevant though (at least not here).

Reply:
The point is just to mix/match various combinations. We test both HTTP and HTTPS earlier and left it as HTTPS in this case. Most people should be using HTTPS anyway :-). You really want this changed to HTTP?

Reply:
I don't mind which scheme is used (it's not relevant to this test), just want the tests to be named accordingly to what they are actually testing (i.e. if we have a test that remarks one scheme then I'd expect other tests checking the outcome of other scheme(s)). Given that removing the scheme from the name is probably better.

Reply:
OK. I buy that. Will do.